### PR TITLE
feat(replays): add replay_id uuid processors to config

### PR DESCRIPTION
**Pull request by JoshFerge**
split out from https://github.com/getsentry/snuba/pull/5787

### CODE REVIEW

File: snuba/datasets/configuration/transactions/storages/transactions.yaml [hunk 1]
```diff
@@ -157,6 +157,7 @@ schema:
       },
       { name: app_start_type, type: String },
       { name: profile_id, type: UUID, args: { schema_modifiers: [nullable] } },
+      { name: replay_id, type: UUID, args: { schema_modifiers: [nullable] } },
```
Comment (aliu3ntry) [inline snuba/datasets/configuration/transactions/storages/transactions.yaml:160]: why is this needed for transactions?

Comment (aliu3ntry) [reply snuba/datasets/configuration/transactions/storages/transactions.yaml:160]: do you also need to add a replay_id column like this to discover.yaml?

File: snuba/datasets/configuration/discover/storages/discover.yaml [hunk 2]
```diff
@@ -88,6 +88,7 @@ schema:
         args: { schema_modifiers: [nullable], size: 64 },
       },
       { name: deleted, type: UInt, args: { size: 8 } },
+      { name: trace_id, type: UUID, args: { schema_modifiers: [nullable] } }
```
Comment (aliu3ntry) [inline snuba/datasets/configuration/discover/storages/discover.yaml:91]: why are we adding trace_id? Isn't this pr a followup to the migration where you added replay_id?

Comment (aliu3ntry) [reply snuba/datasets/configuration/discover/storages/discover.yaml:91]: Do you also have to add a column to the entities/discover.yaml file?

File: snuba/datasets/configuration/transactions/storages/transactions.yaml (hunk 1 above)
Comment (JoshFerge) [reply snuba/datasets/configuration/transactions/storages/transactions.yaml:160]: because we want to query on this column which exists in the transactions dataset

File: snuba/datasets/configuration/discover/storages/discover.yaml (hunk 2 above)
Comment (JoshFerge) [reply snuba/datasets/configuration/discover/storages/discover.yaml]: thanks for the catch, should have been replay_id

**Review by evanh — APPROVED** [review]

**Review by shruthilayaj — APPROVED** [review]
